index.d.ts

```diff
@@ -15,55 +13,53 @@ export type LocaleData = {
   }
 };
 
+type Normalize<Input extends string | number> = Input extends string ? Lowercase<Input> : Input;
 
 
 export function registerLocale(localeData: LocaleData): void;
-export function alpha2ToAlpha3(alpha2: string | Alpha2Code): string;
-export function alpha2ToNumeric(alpha2: string | Alpha2Code): string;
-export function alpha3ToAlpha2(alpha3: string | Alpha3Code): string;
-export function alpha3ToNumeric(alpha3: string | Alpha3Code): string;
-export function numericToAlpha2(numeric: number | string): string;
-export function numericToAlpha3(numeric: number | string): string;
+export function alpha2ToAlpha3<Input extends string>(alpha2: Input): Normalize<Input> extends Normalize<Alpha2Code> ? Alpha3Code : Alpha3Code | undefined;
```

Comment:
But it always returns the upper-cased ones, so we dont normalize the return values